chainer/training/extensions/evaluator.py

```diff
@@ -236,3 +264,41 @@ def finalize(self):
         """
         for iterator in six.itervalues(self._iterators):
             iterator.finalize()
+
+
+class _IteratorProgressBar(util.ProgressBar):
+
+    def __init__(self, iterator, bar_length=None, out=None):
+        if not (hasattr(iterator, 'current_position') and
+                hasattr(iterator, 'epoch_detail')):
+            raise TypeError('iterator must have following attributes: '
```

Comment:
Wording suggestion.
It's better to mention the progress bar.

```suggestion
            raise TypeError('Iterator must have the following attributes to enable a progress bar: '
```